### PR TITLE
Remove Redundant Dipole/Quadrupole Code

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
This code removes `oeprop`'s specialized dipole/quadrupole functions in favor of the general multipole function.

The output for dipoles _does_ change. I can change the printout if there's demand for it, but I don't care either way. I just want to have less code to mess with for the next stage of excited state variable cleanup.

###  Old Function
```
Properties will be evaluated at   0.000000,   0.000000,   0.000000 [a0]

Properties computed using the SCF density matrix

  Nuclear Dipole Moment: [e a0]
     X:     0.0000      Y:     0.0000      Z:     0.9783

  Electronic Dipole Moment: [e a0]
     X:     0.0000      Y:     0.0000      Z:    -0.1680

  Dipole Moment: [e a0]
     X:     0.0000      Y:     0.0000      Z:     0.8103     Total:     0.8103

  Dipole Moment: [D]
     X:     0.0000      Y:     0.0000      Z:     2.0595     Total:     2.0595
```

### New Function
See [here](https://github.com/psi4/psi4/pull/2485#issuecomment-1077774801)

## Checklist
- [x] Quick tests pass
- [x] Properties pass 

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

**Comment by loriab** [comment]
new format lgtm. how about the modifications below:
* add magnitude line
* specify what a.u. is for each L
* any units in square brackets

```
 ------------------------------------------------------------------------------------
     Multipole             Electric (a.u.)       Nuclear  (a.u.)        Total (a.u.)
 ------------------------------------------------------------------------------------

 L = 1.  Multiply by 2.5417464519 to convert [e a0] to [Debye]
 Dipole X            :          0.0000000            0.0000000            0.0000000
 Dipole Y            :          0.0000000            0.0000000            0.0000000
 Dipole Z            :         -0.1680325            0.9783140            0.8102815
 Dipole              :                                                    0.8102815

 ------------------------------------------------------------------------------------
```

File: psi4/src/psi4/libmints/oeprop.cc
Comment (loriab) [inline psi4/src/psi4/libmints/oeprop.cc:1469]: is traceless quadrupole handled by the general multipole code even though non-systematic?

Comment (JonathonMisiewicz) [reply psi4/src/psi4/libmints/oeprop.cc:1469]: Ugh, traceless quadrupole is _not_ handled by the general code. Although I'm not familiar with multipole theory, it looks like traceless multipoles are defined for all orders beyond dipole... I am seriously considering spending the weekend reading up so I have an idea of how to implement this in general.

**Comment by JonathonMisiewicz** [comment]
Format modification looks fine by me. I'll hold off on implementing it until we've had time for other opinions to come in.

**Review by maxscheurer — APPROVED** [review]
LGTM if you add the "hard-coded" computation of traceless quadrupole moments 👍

**Comment by andysim** [comment]
The conversion to traceless form can be done for any order of multipole, but I'm not really sure it's got a use case at this point.  Quadrupoles in traceless form can be useful for force field definitions.  Our CFMM code uses very high orders of multipoles, but they're converted to spherical harmonics to work with all the efficient machinery we've developed so that doesn't need the traceless definition.

**Comment by JonathonMisiewicz** [comment]
> The conversion to traceless form can be done for any order of multipole, but I'm not really sure it's got a use case at this point. Quadrupoles in traceless form can be useful for force field definitions. Our CFMM code uses very high orders of multipoles, but they're converted to spherical harmonics to work with all the efficient machinery we've developed so that doesn't need the traceless definition.

Max and I have traded Slack messages, and we're agreed on adding traceless quadrupoles so those still exist, but not to bother with the other multipoles for the time being. I'll add that to the PR hopefully today. I've been too busy fixing `ambit` to respond to PR feedback here at my usual pace.

File: psi4/src/psi4/libmints/oeprop.cc
Comment (loriab) [inline psi4/src/psi4/libmints/oeprop.cc:826]:
```suggestion
    if (tasks_.count("QUADRUPOLE")) compute_multipoles(2, false);
    else if (tasks_.count("DIPOLE")) compute_multipoles(1, false);
    if (tasks_.count("TRANSITION_QUADRUPOLE")) compute_multipoles(2, true);
    else if (tasks_.count("TRANSITION_DIPOLE")) compute_multipoles(1, true);
```
The `compute_multipoles` performs all orders up to the requested order, right? In that case, may want to consolidate tasks to avoid duplicate printing.

**Review by loriab — COMMENTED** [review]
format changes and traceless quad are in the next PR? in that case, lgtm

**Comment by JonathonMisiewicz** [comment]
New Example of Output:
```
 ------------------------------------------------------------------------------------
     Multipole             Electric (a.u.)       Nuclear  (a.u.)        Total (a.u.)
 ------------------------------------------------------------------------------------

 L = 1.  Multiply by 2.5417464519 to convert [e a0] to [Debye]
 Dipole X            :          0.0000000            0.0000000            0.0000000
 Dipole Y            :          0.0000000            0.0000000            0.0000000
 Dipole Z            :          0.1008259           -0.0087631            0.0920628
 Magnitude           :                                                    0.0920628

 L = 2.  Multiply by 1.3450342976 to convert [e a0^2] to [Debye ang]
 Quadrupole XX       :        -10.9522359            0.0000000          -10.9522359
 Quadrupole XY       :          0.0000000            0.0000000            0.0000000
 Quadrupole XZ       :          0.0000000            0.0000000            0.0000000
 Quadrupole YY       :        -10.9522359            0.0000000          -10.9522359
 Quadrupole YZ       :         -0.0000000            0.0000000           -0.0000000
 Quadrupole ZZ       :        -79.8310241           65.4914008          -14.3396233
 Traceless XX        :         22.9595961          -21.8304669            1.1291291
 Traceless YY        :         22.9595961          -21.8304669            1.1291291
 Traceless ZZ        :        -45.9191921           43.6609339           -2.2582582
```

File: psi4/src/psi4/libmints/oeprop.cc
Comment (zachglick) [inline psi4/src/psi4/libmints/oeprop.cc:940]: Electronic sound better to my ear than electric, but not a big deal either way
```suggestion
        outfile->Printf("     Multipole            Electronic (a.u.)      Nuclear  (a.u.)        Total (a.u.)\n");
```

**Review by zachglick — APPROVED** [review]
The new print out looks great!

File: psi4/src/psi4/libmints/oeprop.cc
Comment (JonathonMisiewicz) [reply psi4/src/psi4/libmints/oeprop.cc]: Last time I checked, m-poles are inherently electric but not necessarily electronic. Good catch.

**Review by loriab — APPROVED** [review]
lgtm! I had forgotten we were storing extra arrays just to have the psivars in Debye*

**Pull request merged**
